ledger/participant-integration-api/src/test/suite/scala/platform/store/appendonlydao/events/MemorySafeBufferedTransactionsReaderSpec.scala

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.dao.events
+
+import akka.stream.scaladsl.Source
+import com.codahale.metrics.MetricRegistry
+import com.daml.ledger.api.testing.utils.AkkaBeforeAndAfterAll
+import com.daml.ledger.api.v1.transaction_service.{
+  GetTransactionTreesResponse,
+  GetTransactionsResponse,
+}
+import com.daml.ledger.api.v1.transaction.{TransactionTree, Transaction => FlatTransaction}
+import com.daml.ledger.offset.Offset
+import com.daml.lf.data.Ref.Party
+import com.daml.lf.data.Time
+import com.daml.logging.LoggingContext
+import com.daml.metrics.Metrics
+import com.daml.platform.store.cache.EventsBuffer
+import com.daml.platform.store.dao.LedgerDaoTransactionsReader
+import com.daml.platform.store.interfaces.TransactionLogUpdate
+import org.mockito.{ArgumentMatchersSugar, MockitoSugar}
+import org.scalatest.Assertion
+import org.scalatest.flatspec.AsyncFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+import scala.concurrent.Future
+
+class MemorySafeBufferedTransactionsReaderSpec
```

Comment:
This test takes a long time (20-30 seconds) and exerts high memory pressure. It is here only for proving the hypothesis and resolution

- [ ] Remove before merging